docs/pwa/pwa.md

```diff
@@ -13,7 +13,7 @@ You can opt in to PWA features using options in `d2.config.json`. Here are the o
 | `pwa.caching`                           | Object                                                              | Contains several properties to configure offline caching by the service worker; see the definitions of the following properties below.                                                                                                                                                                                                                                                                                                                                                                                                                           |
 | `pwa.caching.omitExternalRequests`      | Boolean                                                             | If **true**, omits requests to external domains from the default caching strategies. If **false** (default), requests to external domains will be cached by the default strategies. Note that _all requests are cached during recording mode_, regardless of this setting.                                                                                                                                                                                                                                                                                       |
-| `pwa.caching.patternsToOmit`            | Array of Strings                                                    | A list of URL patterns to omit from the default caching strategies. These strings will be converted to RegExes to test URLs; if a URL matches one of these patterns, that request will not be cached by the default caching strategies. Note that _all requests are cached during recording mode_, regardless of this setting. When choosing these URL filters, note that it is better to _cache too many things_ than to risk _not caching an important part of the app shell_, so choose your filter patterns accordingly.                                     |
+| `pwa.caching.patternsToOmit`            | Array of RegExps or Strings                                         | A list of URL patterns to omit from the default caching strategies. Strings will be converted to RegExes using `new RegExp(str)` to test URLs. If a URL matches one of these patterns, that request will not be cached by the default caching strategies. Note that _all requests are cached during recording mode_, regardless of this setting. When choosing these URL filters, note that it is better to _cache too many things_ than to risk _not caching an important part of the app shell_, so choose your filter patterns accordingly.                   |
```

Comment:
@KaiVandivier is the conversion of strings to regex done in CRA?

Reply:
It happens in the service worker: https://github.com/dhis2/app-platform/blob/f8e18234c0dfbfd13b57e78879475873ec72f05b/pwa/src/service-worker/utils.js#L46-L48

Reply:
Now that I think about it, I should only convert those patterns from strings to regex once - I'll add that to a related PR

Reply:
Overview of what's happening currently: `d2.config.js` has strings or Regexes => CLI converts any regexes to strings so all the patterns are strings => all patterns passed as strings as env vars => SW parses env vars as strings, then uses `new RegExp(patternString)` to test URLs

Reply:
> Now that I think about it, I should only convert those patterns from strings to regex once - I'll add that to a related PR

Added to [this PR](https://github.com/dhis2/app-platform/pull/611)